### PR TITLE
[Fix] Adds null state for when no skills have been linked to any experiences

### LINKED ISSUE #10198: 🐛 Missing `null` state message when a user has experiences but no experiences with linked skills
## 🐛 Bug

Missing null state message when a user has experiences but no experiences with linked skills.

## 🦋 Expected Behaviour

There should be a message for the null state.

## 📋 Steps to Reproduce

1. As a user with an admin role, navigate to `/admin/users/:user-id/profile#career-timeline-section`
2. Ensure the user's page who is being visited has at least one experience but none of the experiences can have linked skills
3. Click the "By Skills" tab
4. Observe empty space

## 📸 Screenshot

Screen Shot 2024-04-26 at 11 49 02

## 🙋‍♀️ Proposed Solution

Replace `null` with a message in `apps/web/src/components/UserProfile/ExperienceSection.tsx`. Message could be something like: No skills have been linked to any experiences.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/db7cbc07a3aba75a8157b53333a8fe8921b0fd43/apps/web/src/components/UserProfile/ExperienceSection.tsx#L196

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the bug was addressed.

- [ ] message renders for the `null` state

### PR DESCRIPTION
**Pull request by mnigh**
🤖 Resolves #10198.

## 👋 Introduction

This PR adds null state for when no skills have been linked to any experiences.

## 🎁 Bonus

- [Consolidated `No information has been provided` strings](https://github.com/GCTC-NTGC/gc-digital-talent/commit/2e9091f266e53cc2ade9ac7b69ab5c3aeb9270bb) where the only difference was a period (kept the one with correct punctuation and 🪓 the one without.
- [Replaced `div` with `Well` component for no skills have been linked to any experiences null state and no experiences null state](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10336/commits/e55b417efb77f5395309d755d66164dc7e4d002e).

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. As a user with an admin role, navigate to `/admin/users/:user-id/profile#career-timeline-section`
2. Ensure the user's page who is being visited has at least one experience but none of the experiences can have linked skills
3. Click the "By Skills" tab
4. Verify null state message exists

## 📸 Screenshot

Screen Shot 2024-05-10 at 12 01 04

### CODE REVIEW

File: apps/web/src/components/UserProfile/ExperienceSection.tsx [hunk 1]
```diff
@@ -194,7 +194,24 @@ const ExperienceSection = ({
...
             ))}
           </Accordion.Root>
-        ) : null}
+        ) : (
+          <div
+            data-h2-background-color="base(background.dark)"
+            data-h2-border="base(1px solid background.darker)"
+            data-h2-padding="base(x1)"
+            data-h2-radius="base(s)"
+          >
```
Comment (esizer) [inline apps/web/src/components/UserProfile/ExperienceSection.tsx:203]: I think we could use the `Well` component here and it would look similar enough.

**Review by esizer — CHANGES_REQUESTED** [review]
One tiny suggestion. Other than that, it looks good :+1:

File: apps/web/src/components/UserProfile/ExperienceSection.tsx (hunk 1 above)
Comment (mnigh) [reply apps/web/src/components/UserProfile/ExperienceSection.tsx:203]: I was hoping to use the `Well` component but it does not match the other null state for the same group which is not a component in of itself. 🤷 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/63d90c822f8eaf01439d79b2fb9e2b553e24ef95/apps/web/src/components/UserProfile/ExperienceSection.tsx#L218-L223

## Screenshot

Screen Shot 2024-05-09 at 12 42 53

Comment (mnigh) [reply apps/web/src/components/UserProfile/ExperienceSection.tsx]: `Well` component replaces these custom `div`s in [e55b417](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10336/commits/e55b417efb77f5395309d755d66164dc7e4d002e).

**Review by esizer — APPROVED** [review]
LGTM! :+1:

**Pull request merged**